src/components/Questionnaire/QuestionTypes/SymptomQuestion.tsx

```diff
@@ -246,48 +261,71 @@ const SymptomRow = React.memo(function SymptomRow({
                   </span>
                 )}
               </Button>
-            </PopoverTrigger>
-            <PopoverContent className="p-0 w-auto" align="start">
-              <Tabs
-                value={activeTab}
-                onValueChange={(v) =>
-                  setActiveTab(v as "absolute" | "relative")
-                }
-              >
-                <TabsList className="grid w-full grid-cols-2">
-                  <TabsTrigger value="absolute">
-                    {t("absolute_date")}
-                  </TabsTrigger>
-                  <TabsTrigger value="relative">
-                    {t("relative_date")}
-                  </TabsTrigger>
-                </TabsList>
-                <TabsContent value="absolute" className="p-0">
-                  <Calendar
-                    mode="single"
-                    selected={
-                      symptom.onset?.onset_datetime
-                        ? new Date(symptom.onset.onset_datetime)
-                        : undefined
-                    }
-                    onSelect={(date: Date | undefined) => {
-                      handleDateChange(date);
-                    }}
-                  />
-                </TabsContent>
-                <TabsContent value="relative" className="p-0">
-                  <RelativeDatePicker
-                    value={
-                      symptom.onset?.onset_datetime
-                        ? new Date(symptom.onset.onset_datetime)
-                        : undefined
-                    }
-                    onDateChange={(date) => handleDateChange(date)}
-                  />
-                </TabsContent>
-              </Tabs>
-            </PopoverContent>
-          </Popover>
+            </DialogTrigger>
+            <DialogPortal>
+              <DialogOverlay className="backdrop-blur-sm" />
+              <DialogContent className="p-0 max-w-md">
+                <div className="flex justify-between items-center p-2 border-b">
+                  <h3 className="font-medium">{t("select_date")}</h3>
+                  <DialogClose asChild>
+                    <Button variant="ghost" size="icon" className="h-8 w-8">
+                      <X className="h-4 w-4" />
```

Comment:
preferred way of working with tailwind v4; tailwind-merge also now supports merging size with h/w classes as well;

```suggestion
                      <X className="size-4" />
```